lms.rb

```
@@ -0,0 +1,286 @@
+
+    when "7"
+      puts "Enter your id"
+      id = gets.chomp.to_i
+      lms.export_student_report(id)
+
+    when "exit"
+      puts "Leaving the program"
+      break
+
+    else
+      puts "Invalid entry. Please retry "
+    end
+  end
+end
+
+#defined a class Student that will store relevant details regarding student and add subjects through subject instance.
+class Student
+ attr_accessor :id, :name, :age, :subjects
+
+  def initialize(id,name,age)
+    @id=id
+    @name=name
+    @age=age
+    @subjects=[]
+  end
+
+  def add_subject(subject)
+    subjects << subject
+  end
+
+end
+
+#defined a class subject that will store relevant details regarding subject detailss
+class Subject
+  attr_accessor :name, :student_id, :score
+  def initialize(student_id,name,score)
+    @student_id=student_id
+    @name=name
+    @score=score
+  end
+end
+
+class LMS
+  attr_accessor :students
+  def initialize
+    @students = {} #initialized student hash
+  end
+
+  # in this function we passed the file_path, kept headers true so that we can directly get its data from headers rather than indexes.
+  def load_students(file_path)
+    CSV.foreach(file_path, headers: true) do |line|
+
+      student_id=line['id'].to_i
+      name=line['name']
+      age=line['age'].to_i
+
+      @students[student_id] = Student.new(student_id,name,age)
+      # stored student data in hash of user by keeping its id as key
+
+    end
+  end
+
+  # in this function we passed the file_path, kept headers true so that we can directly get its data from headers rather than indexes.
+  # or else we can use FILE to open it and separate by delimeters nd store them.
+  def load_subjects(file_path)
+    CSV.foreach(file_path, headers: true) do |line|
+      student_id=line['student_id'].to_i
+      subject_name=line['subject_name']
+      score=line['score'].to_i
+      if @students[student_id]
+      subject=Subject.new(student_id,subject_name,score)
+      @students[student_id].add_subject(subject) 
```

Need to raise an error or implement functionality if the student ID is not present in the @students hash.